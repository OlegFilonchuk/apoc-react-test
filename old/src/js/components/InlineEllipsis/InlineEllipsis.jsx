import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './InlineEllipsis.less';

export default class InlineEllipsis extends React.PureComponent {
  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.PropTypes.node,
    width: PropTypes.string
  };

  static defaultProps = {
    className: '',
    children: '',
    width: 'auto'
  };

  render() {
    const finalClassName = classNames('inline-ellipsis', this.props.className, {
      'inline-ellipsis__with-limited-width': this.props.width !== 'auto'
    });

    return (
      <div className={finalClassName} style={{ width: this.props.width }}>
        {this.props.children}
      </div>
    );
  }
}
