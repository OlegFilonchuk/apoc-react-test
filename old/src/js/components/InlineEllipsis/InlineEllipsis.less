.inline-ellipsis {
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  display: inline-block;
  max-width: 100%;
  vertical-align: middle;

  &__with-limited-width {
    text-align: left;
  }
}
