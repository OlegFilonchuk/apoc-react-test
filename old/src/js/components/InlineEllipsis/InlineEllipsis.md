## Example of InlineEllipsis

    (
      <InlineEllipsis>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab amet consectetur cum cupiditate dicta, eos laboriosam magnam modi molestias nam non quam sint vitae. Accusamus adipisci aspernatur consequatur doloremque eligendi ex fuga fugiat illum ipsa iusto magni nemo nihil nulla perferendis possimus, quasi quibusdam quos reprehenderit sint vero? Adipisci aut cupiditate dignissimos distinctio eligendi esse est fugit hic illo ipsa iste labore laborum libero maxime necessitatibus nulla, pariatur quidem quis quisquam recusandae sequi similique tempore ut veniam vitae. Distinctio eligendi et fugit numquam optio possimus quas sunt! Consequuntur, dignissimos doloribus dolorum explicabo laborum libero magnam maiores necessitatibus nesciunt rem sed.
      </InlineEllipsis>
    )
    
## Example of InlineEllipsis with limited width

    (
      <InlineEllipsis width="100px">
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab amet consectetur cum cupiditate dicta, eos laboriosam magnam modi molestias nam non quam sint vitae. Accusamus adipisci aspernatur consequatur doloremque eligendi ex fuga fugiat illum ipsa iusto magni nemo nihil nulla perferendis possimus, quasi quibusdam quos reprehenderit sint vero? Adipisci aut cupiditate dignissimos distinctio eligendi esse est fugit hic illo ipsa iste labore laborum libero maxime necessitatibus nulla, pariatur quidem quis quisquam recusandae sequi similique tempore ut veniam vitae. Distinctio eligendi et fugit numquam optio possimus quas sunt! Consequuntur, dignissimos doloribus dolorum explicabo laborum libero magnam maiores necessitatibus nesciunt rem sed.
      </InlineEllipsis>
    )
