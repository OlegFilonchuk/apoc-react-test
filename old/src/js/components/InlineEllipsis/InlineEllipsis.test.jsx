import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';

import InlineEllipsis from './InlineEllipsis';

describe('InlineEllipsis', () => {
  let errorConsole;
  let component;

  beforeAll(() => {
    errorConsole = spyOn(console, 'error');
  });

  describe('with default set of props', () => {
    beforeEach(() => {
      component = mount(<InlineEllipsis>Hello world</InlineEllipsis>);
    });

    it('matches snapshot', () => expect(mountToJson(component)).toMatchSnapshot());
  });

  it('prints no errors from propTypes validation', () => {
    expect(errorConsole).not.toHaveBeenCalled();
  });
});
