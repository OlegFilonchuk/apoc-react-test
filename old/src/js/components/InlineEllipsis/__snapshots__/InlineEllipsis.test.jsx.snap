// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineEllipsis with default set of props matches snapshot 1`] = `
<InlineEllipsis
  className=""
  width="auto"
>
  <div
    className="inline-ellipsis"
    style={
      Object {
        "width": "auto",
      }
    }
  >
    Hello world
  </div>
</InlineEllipsis>
`;
